cognite/neat/_rules/transformers/_converters.py

```diff
@@ -1388,9 +1394,38 @@ def as_dms_rules(
             and (prop.property_ == "endNode" or prop.property_ == "end_node")
             and isinstance(prop.value_type, ClassEntity)
         }
-
+        cognite_concepts = self._get_cognite_concepts()
+        cognite_properties: dict[tuple[ClassEntity, str], DMSProperty] = {}
+        ref_cognite_containers: dict[ContainerEntity, DMSContainer] = {}
+        ancestors_by_view: dict[ViewEntity, set[ViewEntity]] = {}
+        if cognite_concepts:
+            if self.client is None:
+                raise CDFMissingClientError(
+                    f"Cannot convert {self.rules.metadata.as_data_model_id()}. Missing Cognite Client."
+                    f"This is required as the data model is referencing cognite concepts in the implements"
+                    f"{humanize_collection(cognite_concepts)}"
+                )
+            cognite_rules = self._get_cognite_properties(cognite_concepts, self.client)
+            for dms_prop in cognite_rules.properties:
+                cognite_properties[(dms_prop.view.as_class(), dms_prop.view_property)] = dms_prop
+            ref_cognite_containers = {container.container: container for container in cognite_rules.containers or []}
+            ancestors_by_view = RulesAnalysis(dms=cognite_rules).implements_by_view(
+                include_ancestors=True, include_different_space=True
+            )
+        parents_by_class = RulesAnalysis(self.rules).parents_by_class(
+            include_ancestors=True, include_different_space=True
+        )
+        for cls_, parents in parents_by_class.items():
+            view_type = cls_.as_view_entity(default_space, default_version)
+            parent_views = {parent.as_view_entity(default_space, default_version) for parent in parents}
+            if view_type in ancestors_by_view:
+                ancestors_by_view[view_type].update(parent_views)
+            else:
+                ancestors_by_view[view_type] = parent_views
         properties_by_class: dict[ClassEntity, list[DMSProperty]] = defaultdict(list)
         referenced_containers: dict[ContainerEntity, Counter[ClassEntity]] = defaultdict(Counter)
+        cognite_containers: dict[ContainerEntity, DMSContainer] = {}
```

Comment:
what is the difference between `cognite_containers` and `ref_cognite_containers` in this scope?